apps/omg_watcher/test/exit_processor/persistence_test.exs

```diff
@@ -0,0 +1,240 @@
+
+  The aim of this test is to ensure, that whatever state the processor ends up being in will be revived from the DB
+  """
+
+  use ExUnitFixtures
+  use OMG.DB.Case, async: true
+
+  alias OMG.API.DevCrypto
+  alias OMG.API.State.Transaction
+  alias OMG.API.Utxo
+  alias OMG.Watcher.ExitProcessor.Core
+
+  require Utxo
+
+  @eth OMG.API.Crypto.zero_address()
+  @not_eth <<1::size(160)>>
+  @zero_address OMG.API.Crypto.zero_address()
+
+  @utxo_pos1 Utxo.position(1, 0, 0)
+  @utxo_pos2 Utxo.position(1_000, 0, 1)
+
+  setup %{db_pid: db_pid} do
+    :ok = OMG.DB.initiation_multiupdate(db_pid)
+  end
+
+  deffixture processor_empty() do
+    {:ok, empty} = Core.init([], [], [])
+    empty
+  end
+
+  deffixture exits(alice) do
+    {[
+       %{amount: 10, currency: @eth, owner: alice.addr, utxo_pos: Utxo.Position.encode(@utxo_pos1), eth_height: 2},
+       %{amount: 9, currency: @not_eth, owner: alice.addr, utxo_pos: Utxo.Position.encode(@utxo_pos2), eth_height: 4}
+     ], [{alice.addr, @eth, 10}, {@zero_address, @eth, 10}]}
+  end
+
+  @tag fixtures: [:processor_empty, :exits]
+  test "persist finalizations with mixed validities",
+       %{processor_empty: processor, db_pid: db_pid, exits: {exit_events, statuses}} do
+    processor
+    |> persist_new_exits(exit_events, statuses, db_pid)
+    |> persist_finalize_exits({[@utxo_pos1], [@utxo_pos2]}, db_pid)
+  end
+
+  @tag fixtures: [:processor_empty, :exits]
+  test "persist finalizations with all valid",
+       %{processor_empty: processor, db_pid: db_pid, exits: {exit_events, statuses}} do
+    processor
+    |> persist_new_exits(exit_events, statuses, db_pid)
+    |> persist_finalize_exits({[@utxo_pos1, @utxo_pos2], []}, db_pid)
+  end
+
+  @tag fixtures: [:processor_empty, :exits]
+  test "persist finalizations with all invalid",
+       %{processor_empty: processor, db_pid: db_pid, exits: {exit_events, statuses}} do
+    processor
+    |> persist_new_exits(exit_events, statuses, db_pid)
+    |> persist_finalize_exits({[], [@utxo_pos1, @utxo_pos2]}, db_pid)
+  end
+
+  @tag fixtures: [:processor_empty, :exits]
+  test "persist challenges and challenge responses",
+       %{processor_empty: processor, db_pid: db_pid, exits: {exit_events, statuses}} do
+    processor
+    |> persist_new_exits(exit_events, statuses, db_pid)
+    |> persist_challenge_exits([@utxo_pos1], db_pid)
+    # NOTE: this might break when respond_to_in_flight_exits_challenges is actually implemented, it works because noop
+    |> persist_respond_to_in_flight_exits_challenges([@utxo_pos1], db_pid)
+  end
+
+  @tag fixtures: [:processor_empty, :exits]
+  test "persist multiple challenges and challenge responses",
+       %{processor_empty: processor, db_pid: db_pid, exits: {exit_events, statuses}} do
+    processor
+    |> persist_new_exits(exit_events, statuses, db_pid)
+    |> persist_challenge_exits([@utxo_pos2, @utxo_pos1], db_pid)
+    # NOTE: see above comment
```

Comment:
Do you mean this comment?:
```
    # NOTE: this might break when respond_to_in_flight_exits_challenges is actually implemented, it works because noop

```
I think it is better to repeat comment than reference comment like this because in some point maybe someone delete above comment without deleting this one.